addons/mail/static/src/new/public/discuss_public.js

```diff
@@ -14,14 +14,38 @@ export class DiscussPublic extends Component {
 
     setup() {
         this.messaging = useMessaging();
+        this.threadService = useState(useService("mail.thread"));
         this.store = useStore();
-        this.state = useState({ welcome: true });
+        this.state = useState({
+            welcome: this.props.data.discussPublicViewData.shouldDisplayWelcomeViewInitially,
+        });
+        useEffect(() => {
+            !this.state.welcome &&
+                this.threadService.fetchChannelMembers(thread) &&
+                this.switchToThreadView();
+        });
```

Comment:
Missing 2nd parameter, which should be a function to return list of values (dependencies)